tests/unit/TransactionPreviewUtils.test.ts

```diff
@@ -0,0 +1,75 @@
+import {buildOptimisticIOUReport, buildOptimisticIOUReportAction} from '@libs/ReportUtils';
+import {createTransactionPreviewConditionals, getTransactionPreviewTextAndTranslationPaths} from '@libs/TransactionPreviewUtils';
+import {buildOptimisticTransaction} from '@libs/TransactionUtils';
+import CONST from '@src/CONST';
+
+const basicProps = {
+    iouReport: buildOptimisticIOUReport(123, 234, 1000, '1', 'USD'),
+    transaction: buildOptimisticTransaction({
+        transactionParams: {
+            amount: 100,
+            currency: 'USD',
+            reportID: '1',
+            comment: '',
+            attendees: [],
+            created: '2024-01-01',
+        },
+    }),
+    translate: jest.fn().mockImplementation((key: string) => key),
+    action: buildOptimisticIOUReportAction('create', 100, 'USD', '', [], '1', undefined, '1'),
+    violations: [],
+    transactions: {},
+    isBillSplit: false,
+    shouldShowRBR: false,
+    isReportAPolicyExpenseChat: false,
+    areThereDuplicates: false,
+};
+
+describe('TransactionPreviewUtils', () => {
```

Comment:
I think this is a good start but there is way more logic to get covered in the tests imho